include/prometheus/exposer.h

```diff
@@ -25,7 +26,12 @@ class Exposer {
 
  private:
   std::unique_ptr<CivetServer> server_;
-  std::vector<std::weak_ptr<Collectable>> collectables_;
+  // \todo maybe better create here class with mutex?
+  //       maybe best solution will be lock free list
+  //       But in any case container need syncronization defence
+  //       One thread can do RegisterCollectable and another thread in server can do collect from container
+  std::mutex collectables_guard_;
+  std::list<std::weak_ptr<Collectable>> collectables_;
```

Comment:
Is there any reason you changed the container from a vector to a list?

Reply:
Reason was delete element when weak_ptr.lock return null shared_pointer, but i see below comment. If exposer will register container only in constructor  so vector is good. I will do fix and will change commit.